plugins/snopes.py

```diff
@@ -1,34 +1,67 @@
 import re
+from urllib import urlencode
+import json
 
 from util import hook, http
 
+SNOPES_AGENT = 'Algolia for vanilla JavaScript (lite) 3.21.1;instantsearch.js 1.11.7;JS Helper 2.19.0'
+SNOPES_API_KEY = '7da15c5275374261c3a4bdab2ce5d321'
+SNOPES_APPLICATION_ID = 'YFRDX308ZD'
 
-search_url = "http://search.atomz.com/search/?sp_a=00062d45-sp00000000"
+ALGOLIA_SEARCH_URL = 'https://{app_id}-dsn.algolia.net/1/indexes/*/queries'
 
 
-@hook.command
-def snopes(inp):
-    ".snopes <topic> -- searches snopes for an urban legend about <topic>"
+def search_snopes(query):
+    search_params = {
+        'query': query,
+        'hitsPerPage': 1,
+        'page': 0
+    }
+
+    query_params = {
+        'x-algolia-agent': SNOPES_AGENT,
+        'x-algolia-api-key': SNOPES_API_KEY,
+        'x-algolia-application-id': SNOPES_APPLICATION_ID
+    }
+
+    post_data = {
+        "requests": [
+            {
+                "indexName": "wp_live_searchable_posts",
+                "params": urlencode(search_params)
+            }
+        ]
+    }
+
+    print(json.dumps(post_data))
 
-    search_page = http.get_html(search_url, sp_q=inp, sp_c="1")
-    result_urls = search_page.xpath("//a[@target='_self']/@href")
+    try:
+        result = http.get_json(
+            ALGOLIA_SEARCH_URL.format(app_id=SNOPES_APPLICATION_ID),
+            query_params=query_params,
+            post_data=json.dumps(post_data)
+        )
+    except http.HTTPError:
+        return None
 
-    if not result_urls:
-        return "no matching pages found"
+    result = result['results'][0]
```

Comment:
can we not return "no matching pages found"?

Reply:
I think you mean that you _would_ like that returned.  

I just pushed a commit so that message is returned in the case where there were no hits, eg, no matches.

There should be no cases where 0 `results` arrays are returned, since it returns 1 result per query.